ui/app/controllers/jobs/index.js

```diff
@@ -342,4 +303,254 @@ export default class JobsIndexController extends Controller {
+    // If it doesnt start with and end with (), or if it does but not all entries are the same propname, or not all entries have == operators, populate them into the searchbox
+
+    filterParts.forEach((part) => {
+      let matched = false;
+      if (part.startsWith('(') && part.endsWith(')')) {
+        part = part.slice(1, -1); // trim the parens
+        // Check to see if the property name (first word) is one of the ones for which we have a dropdown
+        let propName = part.split(' ')[0];
+        if (this.filterFacets.find((facet) => facet.label === propName)) {
+          // Split along "or" and check that all parts have the same propName
+          let facetParts = part.split(' or ');
+          let allMatch = facetParts.every((facetPart) =>
+            facetPart.startsWith(propName)
+          );
+          let allEqualityOperators = facetParts.every((facetPart) =>
+            facetPart.includes('==')
+          );
+          if (allMatch && allEqualityOperators) {
+            // Set all the options in the dropdown to checked
+            this.filterFacets.forEach((group) => {
+              if (group.label === propName) {
+                group.options.forEach((option) => {
+                  set(option, 'checked', facetParts.includes(option.string));
+                });
+              }
+            });
+            matched = true;
+          }
+        }
+      }
+      if (!matched) {
+        unmatchedFilters.push(part);
+      }
+    });
+
+    // Combine all unmatched filter parts into the searchText
+    this.searchText = unmatchedFilters.join(' and ');
+  }
+
+  @computed(
+    'filterFacets',
+    'nodePoolFacet.options.@each.checked',
+    'searchText',
+    'statusFacet.options.@each.checked',
+    'typeFacet.options.@each.checked'
+  )
+  get computedFilter() {
+    let parts = this.searchText ? [this.searchText] : [];
+    this.filterFacets.forEach((group) => {
+      let groupParts = [];
+      group.options.forEach((option) => {
+        if (option.checked) {
+          groupParts.push(option.string);
+        }
+      });
+      if (groupParts.length) {
+        parts.push(`(${groupParts.join(' or ')})`);
+      }
+    });
+    return parts.join(' and ');
+  }
+
+  @action
+  toggleOption(option) {
+    set(option, 'checked', !option.checked);
+    this.updateFilter();
+  }
+
+  // Radio button set
+  @action
+  toggleNamespaceOption(option, dropdown) {
+    this.qpNamespace = option.key;
+    dropdown.close();
+  }
+
+  @action
+  updateFilter() {
+    this.cursorAt = null;
+    this.filter = this.computedFilter;
+  }
+
+  @tracked filter = '';
+  @tracked searchText = '';
+
+  @action resetFilters() {
+    this.searchText = '';
+    this.filterFacets.forEach((group) => {
+      group.options.forEach((option) => {
+        set(option, 'checked', false);
+      });
+    });
+    this.qpNamespace = '*';
+    this.updateFilter();
+  }
+
+  /**
```

Comment:
I'm actually not super thrilled with how searchText feels right now and may reconsider some assumptions later. Right now, if you type a string, it'll prefix that string with `Name contains ______` and it happens a little too quickly for my taste (optimal amount of time it should take my very well be infinity)